interfaces/sourcegen/sourcegen/clib/_Config.py

```diff
@@ -0,0 +1,45 @@
+"""Configuration data class used by CLib source generator."""
+
+# This file is part of Cantera. See License.txt in the top-level directory or
+# at https://cantera.org/license.txt for license and copyright information.
+
+from dataclasses import dataclass
+
+
+@dataclass(frozen=True)
+class Config:
+    """Provides configuration info for the CLibSourceGenerator class"""
+
+    ret_type_crosswalk = {
+        "bool": "int",
+        "void": "int",
+        "int": "int",
+        "size_t": "int",
+        "double": "double",
+        "shared_ptr<T>": "int",
+        "string": "char*",
+        "vector<double>": "double[]",
+        "vector<int>": "int[]",
+    }
+
+    prop_type_crosswalk = {
+        "bool": "int",
+        "int": "int",
+        "size_t": "int",
+        "double": "double",
+        "const double": "double",
+        "double *const": "double*",
+        "const double *const": "const double*",
+        "const string&": "const char*",
+        "shared_ptr<T>": "int",
+        "const shared_ptr<T>": "int",
+        "const vector<double>&": 'const double*',
+        "const vector<shared_ptr<T>>&": 'int[]',
+    }
+
+    includes: dict[str,list[str]]
+
+    @classmethod
+    def from_parsed(cls, *, includes=None) -> 'Config':
```

Comment:
```suggestion
    def from_parsed(cls: Type[_Self], *, includes: dict[str, list[str]] | None = None) -> _Self:
```
and then somewhere above the class definition:
```
_Self = TypeVar("_Self", bound="Config")
```
I think this should be done for all the `classmethod`s in the PR. This (should) allows type checkers to correctly infer the type of a subclass.

The `_Self` type can be replaced by `typing.Self` when we support Python >=3.11 or we can add a dependency on `typing_extensions` and use the `Self` type now, rather than defining the `_Self` type. See: https://stackoverflow.com/a/75337086/2449192

Reply:
While I have adopted all other suggestions, I'd prefer to wait for `typing.Self`, once we drop support for 3.10. Personally, I have never managed to really warm up to `TypeVar`.